gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ClusterStatsJmxMonitor.java

```diff
@@ -0,0 +1,154 @@
+        if (queryResponse != null) {
+            processQueryManagerStats(queryResponse, clusterStats);
+        }
+
+        // Set additional fields
+        clusterStats.proxyTo(backend.getProxyTo())
+                .externalUrl(backend.getExternalUrl())
+                .routingGroup(backend.getRoutingGroup());
+
+        ClusterStats stats = clusterStats.build();
+        log.debug("Completed monitoring for backend: %s. Stats: %s", backend.getProxyTo(), stats);
+        return stats;
+    }
+
+    private void processDiscoveryNodeManagerStats(String response, ClusterStats.Builder clusterStats)
+    {
+        try {
+            JsonNode rootNode = new ObjectMapper().readTree(response);
+            JsonNode attributes = rootNode.get("attributes");
+            if (attributes.isArray()) {
+                for (JsonNode attribute : attributes) {
+                    if ("ActiveNodeCount".equals(attribute.get("name").asText())) {
+                        int activeNodes = attribute.get("value").asInt();
+                        clusterStats.numWorkerNodes(activeNodes)
+                                .trinoStatus((activeNodes > 0) ? TrinoStatus.HEALTHY : TrinoStatus.UNHEALTHY);
+                        log.debug("Processed DiscoveryNodeManager: ActiveNodeCount = %d, Health = %s",
+                                activeNodes, activeNodes > 0 ? "Healthy" : "Unhealthy");
+                        break;
+                    }
+                }
+            }
+        }
+        catch (Exception e) {
+            log.error(e, "Error parsing DiscoveryNodeManager stats");
+        }
+    }
+
+    private void processQueryManagerStats(String response, ClusterStats.Builder clusterStats)
+    {
+        try {
+            JsonNode rootNode = new ObjectMapper().readTree(response);
+            JsonNode attributes = rootNode.get("attributes");
+            if (attributes.isArray()) {
+                int queuedQueries = 0;
+                int runningQueries = 0;
+                for (JsonNode attribute : attributes) {
+                    String name = attribute.get("name").asText();
+                    if ("QueuedQueries".equals(name)) {
+                        queuedQueries = attribute.get("value").asInt();
+                    }
+                    else if ("RunningQueries".equals(name)) {
+                        runningQueries = attribute.get("value").asInt();
+                    }
+                }
+                clusterStats.queuedQueryCount(queuedQueries).runningQueryCount(runningQueries);
+                log.debug("Processed QueryManager: QueuedQueries = %d, RunningQueries = %d",
+                        queuedQueries, runningQueries);
+            }
+        }
+        catch (Exception e) {
+            log.error(e, "Error parsing QueryManager stats");
+        }
+    }
+
+    private String queryJmx(ProxyBackendConfiguration backend, String mbeanName)
```

Comment:
would it be cleaner to use Optional here?

Reply:
Made it optional